exe/driver.cc.in

```diff
@@ -42,6 +42,21 @@ void help(po::options_description& desc)
           "Displays its own version string.", desc) << endl;
 }
 
+void conflicting_options(po::variables_map const& vm, string const& opt1, string const& opt2)
+{
+    if (vm.count(opt1) && !vm[opt1].defaulted() && vm.count(opt2) && !vm[opt2].defaulted()) {
+        throw po::error(opt1 + " and " + opt2 + " options conflict: please specify only one.");
+    }
+}
+
+template<typename... TOpts>
+void conflicting_options(po::variables_map const& vm, string const& opt1, string const& opt2, TOpts && ...opts)
+{
```

Comment:
Well, I'd say stick with your implementation, but the snippet below I think accomplishes the same
goal, without pulling out the "big guns" of variadic function template with recursion. Also, your
implementation produces a much more specific error message, which is definitely good.

void conflicting_options(po::variables_map const& vm, vector<string> opts)
{
  int count = 0;
  for (auto& opt : opt) {
    if (vm.count(opt) && !vm[opt].defaulted())
       count++;
  }
  if (count > 1)
     throw po::error(boost::format("options conflict: please specify only one of %1%.", boost::algorithm::join(opts, ", ")));
}

int
main()
{
    ...
   conflicting_options(vm, {"log-level", "quiet", "verbose", "debug"});
   ...
}